spec/unit/face/plugin_spec.rb

```diff
@@ -2,9 +2,38 @@
 require 'spec_helper'
 require 'puppet/face'
 
-describe Puppet::Face[:plugin, '0.0.1'] do
-  [:download].each do |action|
-    it { is_expected.to be_action action }
-    it { is_expected.to respond_to action }
+describe Puppet::Face[:plugin, :current] do
+
+  let(:pluginface) { described_class }
+  let(:action) { pluginface.get_action(:download) }
+
+  def render(result)
+    action.when_rendering(:console).call(result)
+  end
+
+  context "download" do
+    before :each do
+      Puppet.features.stubs(:external_facts?).returns(true)
```

Comment:
I lots @MikaelSmith comments on an earlier commit, see https://github.com/joshcooper/puppet/commit/54746e7#commitcomment-11389424

I'm proposing we don't add tests for when external_facts? is false, because it never is in practice, and I think we should remove the conditional logic. But do that in a different ticket/PR